inference-engine/src/preprocessing/ie_preprocess_gapi.cpp

```diff
@@ -169,9 +169,50 @@ std::vector<std::vector<cv::gapi::own::Mat>> bind_to_blob(const I420Blob::Ptr& i
     std::vector<std::vector<cv::gapi::own::Mat>> batched_input_plane_mats(batch_size);
     for (int i = 0; i < batch_size; ++i) {
         auto& input = batched_input_plane_mats[i];
-        input.emplace_back(std::move(batched_y_plane_mats[i][0]));
-        input.emplace_back(std::move(batched_u_plane_mats[i][0]));
-        input.emplace_back(std::move(batched_v_plane_mats[i][0]));
+        input.emplace_back(std::move(batched_y_plane_mats[i][0])); //[0] cause only one channel for y
+        input.emplace_back(std::move(batched_u_plane_mats[i][0])); //[0] cause only one channel for u
+        input.emplace_back(std::move(batched_v_plane_mats[i][0])); //[0] cause only one channel for v
+    }
+
+    return batched_input_plane_mats;
+}
+
+std::vector<std::vector<cv::gapi::own::Mat>> bind_to_blob(const BatchedBlob::Ptr& blob,
+                                                          int batch_size) {
+    int underlied_blob_size = 1;
+
+    std::vector<std::vector<cv::gapi::own::Mat>> batched_input_plane_mats(blob->size() * underlied_blob_size);
+    for (size_t i = 0; i < blob->size(); ++i) {
+        auto inBlob = blob->getBlob(i);
+
+        //TODO reuse `underlied_blob_plane` memory
+        std::vector<std::vector<cv::gapi::own::Mat>> underlied_blob_plane;
+
+        if (inBlob->is<NV12Blob>()) {
+            underlied_blob_plane = bind_to_blob(as<NV12Blob>(inBlob), underlied_blob_size);
+        } else if (inBlob->is<I420Blob>()) {
+            underlied_blob_plane = bind_to_blob(as<I420Blob>(inBlob), underlied_blob_size);
+        } else if (inBlob->is<MemoryBlob>()) {
+            //correct batch_size for mem blob
+            std::shared_ptr<MemoryBlob> memBlob = as<MemoryBlob>(inBlob);
+            underlied_blob_size = static_cast<int>(memBlob->getTensorDesc().getDims()[0]);
+            batch_size = underlied_blob_size * blob->size();
+
+            if (batched_input_plane_mats.size() <= static_cast<size_t>(batch_size)) {
+                batched_input_plane_mats.resize(batch_size);
+            } else {
+                IE_THROW() << "fail, invalid dimenstion";
+            }
+
+            underlied_blob_plane = bind_to_blob(memBlob, underlied_blob_size);
+        } /* no alternative here - it had just validated right before */
+
+        // iterate over underlying blob batch_size
+        for (size_t j = 0; j < underlied_blob_plane.size(); j++) {
+            // got  underlying blob components: (y & uv) or (y & u & v)
+            auto &full_component_plane = underlied_blob_plane[j];
+            batched_input_plane_mats.at(i * underlied_blob_size).swap(full_component_plane);
+        }
     }
```

Comment:
This whole method may require some commentary as it is untrivial to get into all this magik

Reply:
agree. This part is aggregation of different Blob processing logic (Memory & NV12,I420) which had legacy-complicated processing code. Comment should have been added